### PR TITLE
Win32: Start of listview scrolling implementation.

### PR DESCRIPTION
**Pull request by madewokherd**

### CODE REVIEW

File: xalia/main.gudl [hunk 1]
```diff
@@ -604,7 +604,7 @@ if (role.(filler or internal_frame or panel or html_container or
               or info_bar or page_tab or page_tab_list or menu_bar or menu or
               combo_box or menu_item or pane or group or document or tree_table or
               toggle_button or list_box or list or table or title_bar or
-              tree)) {
+              tree or icon or small_icon or tile)) {
     recurse: true;
```
Comment (Conmanx360) [inline xalia/main.gudl:608]: These are not necessarily the roles of an element itself, but a particular type of a win32 listview element, correct? I could see this being an issue for other back ends, i.e at-spi2 has a `ATSPI_ROLE_ICON` which I would assume should have the role `icon`, but is probably not considered a container.

File: xalia/Win32/Win32Connection.cs [hunk 2]
```diff
@@ -136,7 +136,19 @@ internal UiDomElement CreateElement(IntPtr hwnd, int idObject = OBJID_CLIENT, in
...
+                        element.AddProvider(scroll_provider);
+
+                        // Apply any customizations
+                        var scrollable_provider = hwnd_ancestor.Element.ProviderByType<IWin32Scrollable>();
+                        if (!(scrollable_provider is null))
+                        {
+                            var custom_provider = scrollable_provider.GetScrollBarProvider(scroll_provider);
+                            if (!(custom_provider is null))
+                                element.AddProvider(custom_provider, 0);
+                        }
```
[the remark this answers is not in the document]
Comment (madewokherd) [reply xalia/Win32/Win32Connection.cs:150]: Yes.

File: xalia/main.gudl (hunk 1 above)
Comment (madewokherd) [reply xalia/main.gudl:608]: Yeah, the role should probably be icon_view or something.

File: xalia/Win32/Win32Connection.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (Conmanx360) [reply xalia/Win32/Win32Connection.cs:150]: Cool, just wanted to make sure. :)

**Comment by Conmanx360** [comment]
Aside from the icon role issue, this looks good to me.

File: xalia/main.gudl (hunk 1 above)
Comment (madewokherd) [reply xalia/main.gudl:608]: Looking at CrossOver in Accerciser, it seems it uses layered_pane for the container and icon for the items.

Comment (madewokherd) [reply xalia/main.gudl]: I'm not arguing, just looking for another container role I can use. I ended up changing the roles in the list view provider itself. How do you feel about it now.

Comment (Conmanx360) [reply xalia/main.gudl]: Looks good to me. :)

**Pull request merged**